### PR TITLE
feat(*/*): Refactor RecordLinkAgent and add build tools

### PR DESCRIPTION
**Pull request by mrogers-5s**
*  Rework core RecordLinkAgent functionality and consolidate repeated code
*  Remove unused includes/definitions
*  Rename table/model classes for external records
*  Rename the `uid` field to `external_id`
*  Add additional documentation to clarify agent option usage
*  Add deploy tools

### CODE REVIEW

File: push.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const { exec } = require('child_process');
+const version = require('./package.json').version;
+
+dir = exec(`gem push huginn_record_link_agent-${version}.gem`, (err, stdout, stderr) => {
+  if (err) {
+    console.log(err);
```
Comment (Ken-Michalak) [inline push.js]: Usually, you don't use `console.log` directly. Does this project use winston or something yet? You should at least be using the levels like `console.error` or whatever it is.

File: lib/huginn_record_link_agent/record_link_agent.rb [hunk 2]
```diff
@@ -393,53 +368,42 @@ def find_source_links(event, data)
...
-                  message: "No match",
-                  target_system: target_system,
-                  target_type: target_type,
-                  target_id: target_id,
-                  source_system: source_system,
-                  source_type: source_type
-                }
-              ]
+
+          payload = create_error_payload(
```
Comment (jspizziri) [inline lib/huginn_record_link_agent/record_link_agent.rb]: @mrogers-5s I'd rather have this method return the full structure, and be parameterized by the required data as individual arguments. That creates a more strictly defined interface.

File: lib/huginn_record_link_agent/record_link_agent.rb [hunk 3]
```diff
@@ -600,75 +537,99 @@ def build_events(event, data, lookup_record, source_records = nil, target_record
...
-              source_type: lookup_record.model_type,
-              source_id: lookup_record.uid,
-              target_system: t[:system],
-              target_type: t[:type],
-              target_id: t[:uid]
-            }
-          })
 
-          create_event payload: payload
+          record_link = create_record_link_payload(
```
Comment (jspizziri) [inline lib/huginn_record_link_agent/record_link_agent.rb]: @mrogers-5s same thing for `create_record_link_payload`. Use parameters as it better enforces the interface.

Comment (mrogers-5s) [reply lib/huginn_record_link_agent/record_link_agent.rb]: Consider it done. I opted not to do that initially given that there were slight variations in the structure of the output based on what was being requested.

I was concerned consolidating would make the flow of event creation harder to follow in the future. I think I found a decent way to pull it off

**Pull request merged**